.github/workflows/release.yml

```diff
@@ -0,0 +1,45 @@
+name: CI & CD
+
+on:
+  push:
+    branches:
+      - master
+
+env:
+  PYTHONDONTWRITEBYTECODE: 1
+
+jobs:
+  build:
+    name: Build wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-python@v4
+        name: Install Python
+        with:
+          python-version: "3.12"
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
```

Comment:
I'm sure snok is trustworthy but I'd rather not have this if it's not required. Can't one just ``pip install "poetry~=1.2"``? I do this in httpstan without issue, https://github.com/stan-dev/httpstan/blob/main/.github/workflows/main.yml